C error: ';' expected (got ")")

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.2.4 0370d4f, timestamp: 2021-09-09 23:00:54 +0300
**OS:** windows,  Windows 10    몠 v19041 64-ࠧ來

**What did you do?**
```
fn main() {
  mut b := &[5, 6, 7]
  {
    a := [1, 2, 3]
    b = &a
  }
  println(b)
}
```
**What did you expect to see?**
```
&[1, 2, 3]
```
**What did you see instead?**
```
C:/Users/vtere/AppData/Local/Temp/v/test.6985592938850061608.tmp.c:6408: warning: cast between pointer and integer of different size
C:/Users/vtere/AppData/Local/Temp/v/test.6985592938850061608.tmp.c:10489: error: ';' expected (got ")")
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -131,7 +131,7 @@ fn (mut g Gen) array_init(node ast.ArrayInit) {
 	if g.is_shared {
 		g.write('}, sizeof($shared_styp))')
 	} else if is_amp {
-		g.write('), sizeof($array_styp))')
+		g.write(')')
 	}
 }
 

--- a/vlib/v/tests/ref_array_init_test.v
+++ b/vlib/v/tests/ref_array_init_test.v
@@ -1,0 +1,9 @@
+fn test_reference_array_init() {
+	mut b := &[5, 6, 7]
+	{
+		a := [1, 2, 3]
+		b = &a
+	}
+	println(b)
+	assert '$b' == '&[1, 2, 3]'
+}

